### PR TITLE
chore: Use latest kind and k8s release

### PR DESCRIPTION
**Pull request by jsturtevant**
Updating to use more recent versions of kubernetes and kind

### CODE REVIEW

**Comment by jprendes** [comment]
Did kind introduce some breaking changes? 😕

**Comment by devigned** [comment]
Might be related to: https://github.com/kubernetes-sigs/kind/issues/3064

**Comment by jsturtevant** [comment]
I did reproduce this locally, Looking into it

**Comment by jprendes** [comment]
I think the remaining failures are just #347 / #354.

**Comment by jsturtevant** [comment]
the remaining failures are 

```
 create containerd task: failed to start shim: start failed: /usr/local/bin/containerd-shim-wasmer-v1: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /usr/local/bin/containerd-shim-wasmer-v1)\n/usr/local/bin/containerd-shim-wasmer-v1: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /usr/local/bin/containerd-shim-wasmer-v1)\n/usr/local/bin/containerd-shim-wasmer-v1: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.33' not found (required by /usr/local/bin/containerd-shim-wasmer-v1)\n: exit status 1: unknown"
```

I believe this is because of the changes in https://github.com/kubernetes-sigs/kind/releases/tag/v0.19.0 from ubuntu to debian

**Comment by jsturtevant** [comment]
@jprendes I am thinking we might be able to use the cross build stuff you've added to correctly link the glibc version?

**Comment by jprendes** [comment]
I can try moving CI to use `cross` tomorrow.

**Comment by jsturtevant** [comment]
leaving notes on the changes for future self:

- in 1.24+ cgroups default to systemd: https://github.com/kubernetes-sigs/kind/issues/1726, so removed the code to fallback to cgroupfs 
- it is recommended to use the pinned version for kind, see the release notes in https://github.com/kubernetes-sigs/kind/releases/tag/v0.20.0

**Comment by jsturtevant** [comment]
I tried using the docker buildx bake and related scripts but having trouble running them, do we have docs on how they should be run? I took a look at the `.github/workflos/release.yaml` and using similiar commands but not having luck.

```
docker buildx bake bins-cross
....
 => CANCELED [linux/amd64 build  9/11] RUN --mount=type=cache,target=/usr/local/cargo/git/db     --mount=type=cache,target=/usr/local/cargo/registr  45.4s
 => ERROR [linux/amd64->arm64 build  2/11] RUN xx-apt-get install -y pkg-config libsystemd-dev libdbus-glib-1-dev build-essential libelf-dev libsec  12.4s
------
 > [linux/amd64->arm64 build  2/11] RUN xx-apt-get install -y pkg-config libsystemd-dev libdbus-glib-1-dev build-essential libelf-dev libseccomp-dev libclang-dev:
#0 10.32 + apt-get  install -y pkg-config:arm64 libsystemd-dev:arm64 libdbus-glib-1-dev:arm64 build-essential:arm64 libelf-dev:arm64 libseccomp-dev:arm64 libclang-dev:arm64
#0 10.33 Reading package lists...
#0 11.57 Building dependency tree...
#0 12.00 Reading state information...
#0 12.23 Some packages could not be installed. This may mean that you have
#0 12.23 requested an impossible situation or if you are using the unstable
#0 12.23 distribution that some required packages have not yet been created
#0 12.23 or been moved out of Incoming.
#0 12.23 The following information may help to resolve the situation:
#0 12.23
#0 12.23 The following packages have unmet dependencies:
#0 12.31  build-essential:arm64 : Depends: dpkg-dev:arm64 (>= 1.17.11)
#0 12.34 E: Unable to correct problems, you have held broken packages.
```

**Comment by jsturtevant** [comment]
> I can try moving CI to use `cross` tomorrow.

I am also having trouble running these locally since I am using ubuntu.  Will need to come up with solution that works locally too.

**Comment by jsturtevant** [comment]
I was able to get the tests passing using the docker build (without the bake stuff) locally using

```
docker build -o dist/bin/ .
make test/k8s-wasmtime
```

**Comment by jsturtevant** [comment]
that time it worked, but need to rebase to pick up #354

File: .github/workflows/action-test-kind.yml [hunk 1]
```diff
@@ -19,16 +19,11 @@ jobs:
...
-        with:
-          name: containerd-shim-${{ inputs.runtime }}-${{ inputs.os }}
-          path: dist
-      - name: Unpack artifats
+      - name: build specific artifacts
         shell: bash
         run: |
           mkdir -p dist/bin
-          tar -xzf dist/containerd-shim-${{ inputs.runtime }}-${{ inputs.os }}.tar.gz -C dist/bin
+          docker build -o dist/bin/ .
```
Comment (jsturtevant) [inline .github/workflows/action-test-kind.yml:26]: This isn't really what we want, it was just a way to test the correctly linked glibc binary works

**Review by Mossaka — DISMISSED** [review]
lgtm! But maybe we should wait for #359 to merge first?

**Comment by jsturtevant** [comment]
Yes, I'll rebase on that once it is ready

**Comment by Mossaka** [comment]
Need a rebase

**Review by Mossaka — APPROVED** [review]

**Pull request merged**